test/e2e/operator_test.go

```diff
@@ -1769,11 +1780,12 @@ func conditionsMatchExpected(expected, actual map[string]string) bool {
 	return reflect.DeepEqual(expected, filtered)
 }
 
-func waitForClusterOperatorConditions(cl client.Client, conditions ...configv1.ClusterOperatorStatusCondition) error {
+func waitForClusterOperatorConditions(t *testing.T, cl client.Client, conditions ...configv1.ClusterOperatorStatusCondition) error {
 	return wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
 		co := &configv1.ClusterOperator{}
 		if err := cl.Get(context.TODO(), controller.IngressClusterOperatorName(), co); err != nil {
-			return false, err
+			t.Logf("failed to get cluster operator %s: %v", controller.IngressClusterOperatorName(), err)
+			return false, nil
```

Comment:
Can you call `controller.IngressClusterOperatorName()` outside of `wait.PollImmediate()` and use the returned value in `wait.PollImmediate()` instead of calling `controller.IngressClusterOperatorName()` twice?